Randolph-dev/Druid_Partnering_Project_React24k
# Scrum 59 api function to fetch content data from drupal

## Related ticket:
[SCRUM-59](https://druidpartneringproject2024bch.atlassian.net/browse/SCRUM-59?atlOrigin=eyJpIjoiODI0N2VlYzljZjAxNDk2NGIxYWYxYjE4OGI5NmQxMTMiLCJwIjoiaiJ9)

## In this PR:
- Write a reusable function to fetch content data from Drupal
- Use that function to fetch data for all pages
- Change the Nav items, Routes and Routing Components according to the client requirements
- Some refactoring in the NavBar: make a reusable NavLink component

## Testing instruction
- Checkout branch
- Make sure Backend is running, newest db imported and configured imported
- Make sure the .env.local file in root have the right environment path to the running Drupal site
- Make sure the CORS bundle is installed and activated
- Go to frontend page, click on any link, check the console log for the data for that page

## Screenshot
Here is how console should look like when you visit a page
<img width="805" alt="image" src="https://github.com/user-attachments/assets/cf2aeb76-87ca-4c8d-9d5f-04f418e00b1f">


[SCRUM-59]: https://druidpartneringproject2024bch.atlassian.net/browse/SCRUM-59?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
